bftengine/tests/simpleKVBCTests/TesterReplica/main.cpp

```diff
@@ -70,12 +62,9 @@ int main(int argc, char **argv) {
 	rp.replicaId = UINT16_MAX;
 
 	// allows to attach debugger
-	if(rp.debug)

```

Comment:
Why did you remove the shutdown cleanup code? I remember that when I used the library with TCP it may not release the socket and then the consecutive run will fail when binding to the port.

Reply:
The call to r->stop was throwing an exception inside the signal handler. It looked like the asserts in the tests caused memory to be freed during unwind that then were used during stop. Guy said we didn't support graceful shutdown right now and that he didn't think we needed the signal handler in the test so I removed the code rather than digging into that specific error at this time. We still need to look into it, but since we are going to be making some changes to interfaces, it may be easier to do that then. 

I didn't realize the part about the socket. That's likely due to TIME_WAIT state. We can likely get rid of that issue by just setting SO_REUSEADDR on the socket.

Reply:
OK, if it works - I am fine